In [1]:
# libraries
import pandas as pd
import warnings
warnings.filterwarnings("ignore")

C:\Users\Cynthia\AppData\Roaming\Python\Python311\site-packages\pandas\core\arrays\masked.py:60: UserWarning: Pandas requires version '1.3.6' or newer of 'bottleneck' (version '1.3.5' currently installed).
  from pandas.core import (


## DF1

In [2]:
df1=pd.read_excel(r"data\Mexico-real-estate-1.xlsx")
nRows, nCols = df1.shape

print(f"There are {nRows} rows and {nCols} columns.")
print(df1.info()) #looking at df1 datatypes for all the columns.
df1.head()

There are 700 rows and 7 columns.
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 700 entries, 0 to 699
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Unnamed: 0     700 non-null    int64  
 1   property_type  700 non-null    object 
 2   state          700 non-null    object 
 3   lat            583 non-null    float64
 4   lon            583 non-null    float64
 5   area_m2        700 non-null    float64
 6   price_usd      700 non-null    object 
dtypes: float64(3), int64(1), object(3)
memory usage: 38.4+ KB
None


Unnamed: 0 property_type                            state        lat  \
0           1         house                 Estado de México  19.560181   
1           2         house                       Nuevo León  25.688436   
2           3     apartment                         Guerrero  16.767704   
3           4     apartment                         Guerrero  16.829782   
4           5         house  Veracruz de Ignacio de la Llave        NaN   

          lon  area_m2   price_usd  
0  -99.233528    150.0  $67,965.56  
1 -100.198807    186.0  $63,223.78  
2  -99.764383     82.0  $84,298.37  
3  -99.911012    150.0  $94,308.80  
4         NaN    175.0  $94,835.67

* It appears as if there are 700 rows and 7 columns.However, "Unnamed: 0" is an unecessary column that must be removed.Then there will be 2 datatypes present in the dataset, float and object.

* Price column, "price_usd" appears to be of type object, will have to inspect and remove unecessary characters and convert it the correct datatype.

In [3]:
# removing the unnecessary characters and converting the column to a correct datatype
df1.price_usd=df1.price_usd.str.replace("$", "").str.replace(",", "").astype(float)

In [4]:
# checking for missing values
df1.isnull().sum()/len(df1) * 100

Unnamed: 0        0.000000
property_type     0.000000
state             0.000000
lat              16.714286
lon              16.714286
area_m2           0.000000
price_usd         0.000000
dtype: float64

There is 117 latitude and longitude values missing, which is 16.71 percent column data missing.

In [5]:
# dropping nan values for now.
df1.dropna(inplace=True)

In [6]:
# checking for duplicated instances
df1.duplicated(keep=False).sum()

0

There are no duplicates values

In [7]:
# # dropping unnamed column which is not necessary.
df1.drop(columns="Unnamed: 0", inplace=True)
df1.head()

property_type             state        lat         lon  area_m2  price_usd
0         house  Estado de México  19.560181  -99.233528    150.0   67965.56
1         house        Nuevo León  25.688436 -100.198807    186.0   63223.78
2     apartment          Guerrero  16.767704  -99.764383     82.0   84298.37
3     apartment          Guerrero  16.829782  -99.911012    150.0   94308.80
5         house           Yucatán  21.052583  -89.538639    205.0  105191.37

## Df2

In [8]:
df2=pd.read_excel(r"data\Mexico-real-estate-2.xlsx")

nRows, nCols = df2.shape

print(f"There are {nRows} rows and {nCols} columns.")
print(df2.info()) #looking at df1 datatypes for all the columns.
df2.head()

There are 700 rows and 7 columns.
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 700 entries, 0 to 699
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Unnamed: 0     700 non-null    int64  
 1   property_type  700 non-null    object 
 2   state          700 non-null    object 
 3   lat            571 non-null    float64
 4   lon            571 non-null    float64
 5   area_m2        700 non-null    float64
 6   price_mxn      700 non-null    float64
dtypes: float64(4), int64(1), object(2)
memory usage: 38.4+ KB
None


Unnamed: 0 property_type       state        lat         lon  area_m2  \
0           1     apartment  Nuevo León  25.721081 -100.345581     72.0   
1           2     apartment      Puebla        NaN         NaN    190.0   
2           3         house     Morelos  23.634501 -102.552788    360.0   
3           4         house     Morelos        NaN         NaN     76.0   
4           5         house      Puebla        NaN         NaN    200.0   

   price_mxn  
0  1300000.0  
1  2500000.0  
2  5300000.0  
3   820000.0  
4  1100000.0

* It appears as if there are 700 rows and 7 columns.However, "Unnamed: 0" is an unecessary column that must be removed.Then there will be 2 datatypes present in the dataset, float and object.

* Price column, "price_mxn" is in mexican currency, will have to convert an american currency.

In [9]:
# converting 'price_mxn' columns from mxn to usd currency as per other dataframe and assging the result to a new column.
# This data was collected in 2014, a dollar cost 19 pesos then.
df2["price_usd"]=df2.price_mxn/19 

In [10]:
#converting mxn currency to to usd currency.
df2.drop(columns=["price_mxn", "Unnamed: 0"], inplace=True)

In [11]:
# checking for missing values.
df2.isnull().sum()/len(df2) * 100

property_type     0.000000
state             0.000000
lat              18.428571
lon              18.428571
area_m2           0.000000
price_usd         0.000000
dtype: float64

There is 129 latitude and longitude values missing, which is 18.43 percent of the data in these 2 columns.

In [12]:
# dropping nan values for now
df2.dropna(inplace=True)

In [13]:
# checking for duplicated instances
df2.duplicated(keep=False).sum()

2

There are 2 instances that are duplicates of each other.

In [14]:
# dropping duplicated values.
df2=df2.drop_duplicates(keep='first')
df2.head()

property_type             state        lat         lon  area_m2  \
0     apartment        Nuevo León  25.721081 -100.345581     72.0   
2         house           Morelos  23.634501 -102.552788    360.0   
6     apartment  Estado de México  19.272040  -99.572013     85.0   
7         house   San Luis Potosí  22.138882 -100.996510    158.0   
8     apartment  Distrito Federal  19.394558  -99.129707     65.0   

       price_usd  
0   68421.052632  
2  278947.368421  
6   65789.473684  
7  111578.947368  
8   39904.736842

## DF3

In [15]:
df3=pd.read_excel(r"data\Mexico-real-estate-3.xlsx")
nRows, nCols = df3.shape

print(f"There are {nRows} rows and {nCols} columns.")
print(df3.info()) #looking at df1 datatypes for all the columns.
df3.head()

There are 700 rows and 6 columns.
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 700 entries, 0 to 699
Data columns (total 6 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Unnamed: 0               700 non-null    int64  
 1   property_type            700 non-null    object 
 2   place_with_parent_names  700 non-null    object 
 3   lat-lon                  582 non-null    object 
 4   area_m2                  700 non-null    float64
 5   price_usd                700 non-null    float64
dtypes: float64(2), int64(1), object(3)
memory usage: 32.9+ KB
None


Unnamed: 0 property_type  \
0           1     apartment   
1           2         house   
2           3         house   
3           4         house   
4           5     apartment   

                             place_with_parent_names                 lat-lon  \
0  |México|Distrito Federal|Gustavo A. Madero|Acu...     19.52589,-99.151703   
1           |México|Estado de México|Toluca|Metepec|  19.2640539,-99.5727534   
2  |México|Estado de México|Toluca|Toluca de Lerd...    19.268629,-99.671722   
3        |México|Morelos|Temixco|Burgos Bugambilias|                     NaN   
4  |México|Veracruz de Ignacio de la Llave|Veracruz|    19.511938,-96.871956   

   area_m2  price_usd  
0     71.0   48550.59  
1    233.0  168636.73  
2    300.0   86932.69  
3    275.0  263432.41  
4     84.0   68508.67

* There are 700 rows and 6 columns.However, "Unnamed: 0" is an unecessary column that must be removed.Then there will be 5 columns and 2 datatypes present in the dataset, float and object.


<font size="+1">Column 'place_with_parent_names' and 'lat-lon' does not seem to be in a proper order. 

### lat-lon

In [16]:
# splitting lat-lon to get 2 columns.
df3[["lat","lon"]]=df3["lat-lon"].str.split(",", expand=True).astype(float)

### place_with_parent_names

In [17]:
# spliting 'place_with_parent_names' to extract the district column.
df3["state"]=df3.place_with_parent_names.str.split("|", expand=True)[2]

In [18]:
# dropping unnecessary columns
df3.drop(columns=["lat-lon","place_with_parent_names","Unnamed: 0"], inplace=True)

In [19]:
df3.isnull().sum()/ len(df3) * 100

property_type     0.000000
area_m2           0.000000
price_usd         0.000000
lat              16.857143
lon              16.857143
state             0.000000
dtype: float64

There is 118 latitude and longitude values missing, which is 16.86 percent of the data in these 2 columns.

In [20]:
# dropping nan values for now.
df3.dropna(inplace=True)


In [21]:
# checking for duplicate values
df3.duplicated(keep=False).sum()

2

In [22]:
# dropping duplicated values
df3=df3.drop_duplicates(keep='first')

In [23]:
df3=df3.reindex(columns=['property_type', 'state', 'lat', 'lon', 'area_m2','price_usd'])

In [24]:
df3.head()

property_type                            state        lat         lon  \
0     apartment                 Distrito Federal  19.525890  -99.151703   
1         house                 Estado de México  19.264054  -99.572753   
2         house                 Estado de México  19.268629  -99.671722   
4     apartment  Veracruz de Ignacio de la Llave  19.511938  -96.871956   
5         house                          Jalisco  20.689157 -103.366728   

   area_m2  price_usd  
0     71.0   48550.59  
1    233.0  168636.73  
2    300.0   86932.69  
4     84.0   68508.67  
5    175.0  102763.00

## Concatenate

In [25]:
df=pd.concat([df1, df2, df3],ignore_index=True).drop_duplicates()
df.shape

(1730, 6)

## Save to CSV

In [26]:
df.to_csv("data\Mexico-real-estate-clean.csv", index=False)